### PR TITLE
Remove unnecessary validation of Data

### PR DESCRIPTION
**Pull request by michaelpj**
Close to the release of Alonzo, we improved the format of `Data` to
allow arbitrary-size integers and bytestrings by requiring them to be
encoded in CBOR using indefinite-length bytestrings. This means that the
client-side validation performed by the API was never necessary, and can
be removed.

I guess it's too late to backport this to the release branch, but at least it can make it out in the next minor release.

### CODE REVIEW

**Review by dcoutts — APPROVED** [review]
LGTM

**Comment by michaelpj** [comment]
Ugh, mandatory hlint :(

File: cardano-api/src/Cardano/Api/ScriptData.hs [hunk 1]
```diff
@@ -188,16 +188,11 @@ validateScriptData d =
...
-          || n < negate (fromIntegral (maxBound :: Word64))
-        ]
-    collect (ScriptDataBytes bs) =
-        [ ScriptDataBytesTooLong len
-        | let len = BS.length bs
-        , len > scriptDataByteStringMaxLength
-        ]
+    
+    -- Arbitrary size numbers are fine
+    collect (ScriptDataNumber _) = [ ]
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/ScriptData.hs:193]: Small style change: `[]` instead of `[ ]`

**Review by Jimbo4350 — APPROVED** [review]
LGTM, one minor style adjustment and can you squash the commits?

**Review by JaredCorduan — APPROVED** [review]

**Comment by michaelpj** [comment]
Updated.

**Comment by Jimbo4350** [comment]
@michaelpj can you open a branch on `cardano-node` so that github actions get triggered? I don't think I can `bors` this.

**Comment by michaelpj** [comment]
I don't have write permissions.

**Comment by michaelpj** [comment]
@Jimbo4350 can I get write permissions? or what should I do?

**Comment by michaelpj** [comment]
Opened https://github.com/input-output-hk/cardano-node/pull/4344 from this repo.

**Pull request closed without merging**